gemini/use-cases/retail/product_attributes_extraction.ipynb

```diff
@@ -0,0 +1,895 @@
+        "\n",
+        "\n",
+        "def load_bytes_from_uri(image_uri: str) -> bytes:\n",
+        "    \"\"\"Load image bytes from a remote or local file URI.\n",
+        "\n",
+        "    Args:\n",
+        "      image_uri (str): The uri or the local file path for the image.\n",
+        "\n",
+        "    Returns:\n",
+        "      The object/image bytes.\n",
+        "    \"\"\"\n",
+        "    image_bytes = None\n",
+        "    if image_uri.startswith(\"http://\") or image_uri.startswith(\"https://\"):\n",
+        "        response = requests.get(image_uri)\n",
+        "        if response.status_code == 200:\n",
+        "            image_bytes = response.content\n",
+        "        else:\n",
+        "            print(\n",
+        "                f\"Fetch image failed for {image_uri}, status code: {response.status_code}\"\n",
+        "            )\n",
+        "    else:\n",
+        "        image_bytes = open(image_uri, \"rb\").read()\n",
+        "    return image_bytes\n",
+        "\n",
+        "\n",
+        "def load_bytes_from(any_uri: str) -> bytes:\n",
+        "    \"\"\"Load image bytes from a remote or local file URI.\"\"\"\n",
+        "    if any_uri.startswith(\"gs://\"):\n",
+        "        image_bytes = load_bytes_from_gcs(any_uri)\n",
+        "    else:\n",
+        "        image_bytes = load_bytes_from_uri(any_uri)\n",
+        "    return image_bytes\n",
+        "\n",
+        "\n",
+        "def get_mime_from_uri(image_uri: str) -> str:\n",
+        "    \"\"\"Get the mime type from the image uri.\"\"\"\n",
+        "    if image_uri.endswith(\".png\"):\n",
+        "        return \"image/png\"\n",
+        "    elif image_uri.endswith(\".gif\"):\n",
+        "        return \"image/gif\"\n",
+        "    else:\n",
+        "        # Assume JPEG as the default mime\n",
+        "        return \"image/jpeg\"\n",
+        "\n",
+        "\n",
+        "def parse_json_from_markdown(answer: str) -> dict[str, list[str]]:\n",
+        "    \"\"\"Parse the json from the markdown answer.\n",
+        "\n",
+        "    Args:\n",
+        "        answer (str): The markdown answer from the model.\n",
+        "\n",
+        "    Returns:\n",
+        "        A parsed json dictionary}\n",
+        "    \"\"\"\n",
+        "    lines = answer.split(\"```\")\n",
+        "    try:\n",
+        "        # Tries to parse the last json in the answer.\n",
+        "        answer = lines[-2]\n",
+        "        if answer.startswith(\"json\"):\n",
+        "            answer = answer[4:]\n",
+        "        result = json.loads(answer)\n",
+        "    except json.JSONDecodeError:\n",
+        "        # Falls back to the first json in the answer.\n",
+        "        answer = lines[1]\n",
+        "        if answer.startswith(\"json\"):\n",
+        "            answer = answer[4:]\n",
+        "        result = json.loads(answer)\n",
+        "    return result"
```

Comment:
If you use the `response_mime_type` parameter to `GenerationConfig` then you shouldn't need to do this.

Reply:
Oh, the expected response is a mix of (several) json and some text responses in between, and therefore need to parse the json out of it. Not sure "response_mime_type" will help with the parsing.

Reply:
Ok, that makes more sense since everything is included in one response.